src/librustc_ast/expand/allocator.rs

```diff
@@ -9,10 +9,10 @@ pub enum AllocatorKind {
 }
 
 impl AllocatorKind {
-    pub fn fn_name(&self, base: &str) -> String {
+    pub fn fn_name(&self, base: Symbol) -> String {
         match *self {
-            AllocatorKind::Global => format!("__rg_{}", base),
-            AllocatorKind::Default => format!("__rdl_{}", base),
+            AllocatorKind::Global => format!("__rg_{}", base.as_str()),
```

Comment:
This `as_str` should be unnecessary

Reply:
It currently is unnecessary, but I'm planning to do a follow-up in which I remove the `Debug` and `Display` impls for `Symbol`. I want the `as_str()` to be required, so that the cost of the interner lookup isn't hidden. So I made this `as_str()` call explicit.

Reply:
>a follow-up in which I remove the `Debug` and `Display` impls for `Symbol`

Both this and the last commit removing `fn as_str` from `Ident` seem pretty impractical.
`Display` impls are used in a large number of places and `.as_str()` calls will mostly add noise as I see it.

Reply:
I like having `Display` and `Debug` impls. What's the motivation for removing them?

I can kinda get behind the `as_str` removal from `Ident` (since it makes it easier for this PR to find places where it should be matched on symbold), which is why I didn't comment on it. But as long as there is any desire to keep the original system (and this desire is there, as petrochenkov likes it), I don't believe we should remove it. It's not strictly better, even if it really helped this PR.

Reply:
I have updated to remove these two `.as_str()` calls.

Reply:
Can you also revert the `Ident::as_str` removal? It's contentious and I don't have a strong opinion for it. Withough consensus I don't think we should do it. If you want to pursue it, I recommend an MCP

Reply:
I have removed that commit.